src/components/multi-select/multi-select.css

```diff
@@ -0,0 +1,80 @@
+@import '../../styles/colors-v2.css';
+@import '../../styles/spacing.css';
+
+.container {
+  composes: tag-list from '../tag-list/tag-list.css';
+  composes: input-container from '../__internal__/abstract-custom-field/abstract-custom-field.css';
+  height: auto;
+  padding-top: 0;
+  padding-bottom: 0;
+}
+
+.icon-clear {
+  cursor: pointer;
+  padding: var(--spacing-small);
+}
+
+.icon-clear-readonly {
+  composes: icon-clear;
+  cursor: unset;
+  pointer-events: none;
+}
+
+.icon-container {
+  composes: icons from '../form-control-icons/form-control-icons.css';
+  top: calc(var(--spacing-large) + var(--spacing-small));
+}
+
+.input {
+  border: none;
+  flex-grow: 1;
+  flex-shrink: 1;
+  height: var(--spacing-x-large);
+  width: 0;
+}
+
+.input-readonly {
+  composes: input;
+  background-color: transparent;
+}
+
+.form-control-children-container {
+  composes: input-styles from '../__internal__/abstract-custom-field/abstract-custom-field.css';
+  align-items: center;
+  height: auto;
+  padding-right: calc(var(--spacing-x-large) + var(--spacing-medium));
+}
+
+.form-control-children-container-readonly {
+  composes: form-control-children-container;
+  background-color: var(--mds-grey-3);
+}
+
+.popup-container {
+  composes: container from '../listbox/listbox.css';
+  background-color: var(--white);
+  box-sizing: border-box;
+  max-height: 160px;
+  overflow-y: scroll;
+  position: absolute;
+  width: 100%;
+  z-index: 10;
```

Comment:
This is one of those things that I'd _really_ love to have in the design system (z-indicies mappings). Until that happens, I tend to go low, until there's a reason to bump it up. This `z-index` is copied from the existing `CustomFieldInputMultiChoice`, so we know it at least works for all the same use-cases (including the configure modal in Custom Forms).

I'd prefer not to change it to be higher until we have a specific use-case that is exemplifying bad behavior.

Reply:
The Date stuff was probably copy-pasta or incorrect z-index usage tbh.

AFAIK, a dialog would rely on block context where we would force a new component to be in a new "context". This implies that any previous context can _never_ be rendered above the new context. In this sense,

```
<html>
  <body>
    <normal app>
      <popup z-index="over 9000" />
    </normal app>
    <new context>
      <dialog z-index="1" />
    </new context>
  </body>
</html>
```

The normal app's popup would have a z-index of its value + its parent's values. Regardless of the computed value, it can never be "on-top" of the new context -- even if the dialog's computed z-index is lower than the popup's. 

It's some crazy stuff and really annoying at the same time. But it would help us in this case. Last I looked into this kind of stuff was when I was re-creating Gantt in Backbone Marionette so that was like 5 years ago. I wonder if browsers have changed 🤔 

Regardless, I think we're good. We should strive for smaller z-index bumping. Given that, the "max integer for date" sounds like a relatively small mistake.

Reply:
It depends on how the dialog is implemented, but yeah, usually it involves creating a new "stacking context", and like you said, there's no CSS way to make something in one context appear above another context. That said, we're safe in the case where the dropdown is a child of the new context, as the z-index will be correct in that set of children.

See https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Positioning/Understanding_z_index/The_stacking_context